### PR TITLE
docs: Add introduction docs to storybook

### PR DESCRIPTION
**Pull request by cillascillas**
Given the [design proposal of DS documentation](https://www.figma.com/proto/7RhweKibaUnxX6WYBl7zKW/Design-System-Beta?node-id=3385%3A359&viewport=732%2C-783%2C0.3692046105861664&scaling=scale-down-width) implement new introduction section:

- [x] Add `@storybook/addon-docs` package to dev dependencies
- [x] Add Introduction mdx files to a new `docs` folder
- [x] Set order of files in sidebar
- [x] Add actual/verified content provided by the design team
- [x] Write instructions on how to edit content

### To be defined/discussed:

**- `docs` folder has been placed inside `src`:** The initial strategy was to put it at the same level of `src` to make easier for non-developers to find it to edit. However, it would add a new folder to the list of folders to "share" when publishing the package and it feels unnatural to have a folder with `.stories.mdx` outside the source.

**- Ordering/sorting stories:** As done with other sections, it has been added to the preview options in the correct order. It would have been great to avoid this step and make the files order by filename (maybe including numbering in the filenames) but it was discarded in the first place to avoid jeopardising the rest of sorting for components' stories.

**- Images:** Discuss with the design team if they would like to include images to define where to store them or if they need any additional feature.

### CODE REVIEW

**Comment by cillascillas** [comment]
Brand new things after a fruitful Friday afternoon session with @icurdu and @afontcu:

- `docs` folder: Finally moved outside `src` so the design team can easily find them to edit
- `colors.stories.js`: Moved into `docs` folder (since it was not attached to any specific component and it would be edited by the design team as well) 
- `MaAlert.docs.mdx`: First sample of content for new **Docs** tab (next to **Canvas**) also included in `docs` folder and to be edited by the design team with up to date content: [preview here](https://601199d127273f0021b037b0-trnaudgimd.chromatic.com/?path=/docs/components-alert--alert)

**Comment by cillascillas** [comment]
Last additions:

- [ ] Enable **Docs** tabs with text placeholder for all the components
- [ ] Images folder and modify `serve` and `build` commands to make them available

**Review by edufarre — APPROVED** [review]
Noice!! Look promising 🚀

File: docs/introduction/welcome.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- Unit testing based on [Jest](https://jestjs.io/) and [Vue Testing Library](https://github.com/testing-library/vue-testing-library).
+- The design library is built using [Figma](https://www.figma.com/file/7RhweKibaUnxX6WYBl7zKW/Design-System-Beta?node-id=12%3A3).
+
+## Changelog
+
+Margarita follows semantic versioning. In short, this means we use patch versions for bugfixes, minor versions for new features, and major versions for breaking changes. You can see changes here.
+
+## About Holaluz
+
+Holaluz's mission is to connect people to green energy. Our engineers and designers work in Barcelona, Caldes de Montbui, Tarragona, Sant Cugat, and other parts of Spain, Catalonia, and Europe. We write code in Javascript and Jquery.
```
Comment (afontcu) [inline docs/introduction/welcome.stories.mdx:29]: Hahahahaha @icurdu got you!!! :trollface:

Comment (edufarre) [inline docs/introduction/welcome.stories.mdx:29]: I just felt ignored 🤣 . I mean, I reckon we do have a lot (and by saying a lot, I mean a lot) of foggy days in Lleida, BUT this is not enough to not be part of the map 🎭  🌫️ 🌁

File: docs/introduction/getting-started.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+### **Installing Margarita locally**
+
+[https://github.com/holaluz/margarita#installing-margarita-locally](https://github.com/holaluz/margarita#installing-margarita-locally)
+
+#### Deployment and versioning
+
+[https://github.com/holaluz/margarita#deployment-and-versioning](https://github.com/holaluz/margarita#deployment-and-versioning)
+
+Please, reach out to us via Slack in our #design-system channel if you have any doubt or inquiry. We will be more than happy to help you 🤗
```
Comment (edufarre) [inline docs/introduction/getting-started.stories.mdx:26]: Same. I would remove this sentence for now, until we do discuss to whom is this documentation directed. (in-house or third-party users)

File: docs/introduction/welcome.stories.mdx (hunk 1 above)
Comment (edufarre) [reply docs/introduction/welcome.stories.mdx:29]: hahahahahah Dammit, that was a good one 🤣

**Review by afontcu — COMMENTED** [review]
Looks amazing 🤩 left some minor comments here and there

File: docs/introduction/welcome.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+[![Version](https://img.shields.io/npm/v/@holaluz/margarita)](https://www.npmjs.com/package/@holaluz/margarita)
+
+- Built on top of [@vue/cli](https://cli.vuejs.org/).
+- Uses [Storybook](https://github.com/storybooks/storybook) to develop, showcase, and test each component.
+- Unit testing based on [Jest](https://jestjs.io/) and [Vue Testing Library](https://github.com/testing-library/vue-testing-library).
+- The design library is built using [Figma](https://www.figma.com/file/7RhweKibaUnxX6WYBl7zKW/Design-System-Beta?node-id=12%3A3).
+
+## Changelog
+
+Margarita follows semantic versioning. In short, this means we use patch versions for bugfixes, minor versions for new features, and major versions for breaking changes. You can see changes here.
```
[the remark this answers is not in the document]
Comment (edufarre) [reply docs/introduction/welcome.stories.mdx:25]: wouldn't be: see `the changelog` tho?

[the remark this answers is not in the document]
Comment (afontcu) [reply docs/introduction/welcome.stories.mdx:25]: hmmm sure, updated.

And actually we should mention kalimotxo, too!

**Comment by carlotamolintr** [comment]
> * `MaAlert.docs.mdx`: First sample of content for new **Docs** tab (next to **Canvas**) also included in `docs` folder and to be edited by the design team with up to date content: [preview here](https://601199d127273f0021b037b0-trnaudgimd.chromatic.com/?path=/docs/components-alert--alert)

how did you get the preview link? Is a branch link, right?

Related to all the job, it looks super nice! :)

**Review by carlotamolintr — APPROVED** [review]

**Comment by afontcu** [comment]
> how did you get the preview link? Is a branch link, right?

As mentioned in https://github.com/holaluz/margarita/pull/337, now each margarita PR creates a custom version of storybook with its changes, so we can easily check the result. Accessible here:

image

File: docs/introduction/getting-started.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - **In Margarita:** components ready to use in both design and front-end.
+    - **In Kalimotxo:** components with Holaluz's Business Logic.
+    - **In Component Parking:** Components that are used ad-hoc in particular projects and are waiting to be drafted as Margarita components. **Use these components exceptionally, because they're not implemented in Margarita yet and availability depends on the project you are working on.**
+- **Library components**: These components are only used to maintain the library visually. Do not use them for building interfaces.
+
+### Adding more components in the library
+
+At the moment, we are not allowing external users to contribute to our design system with new components or improvements as we are still creating a stable version and finding the most efficient way to contribute to our design flow. 
+
+Nevertheless, feel free to comment in our Figma file or share with us anything you want via our Slack Channel #design-system or #product-design
```
[the remark this answers is not in the document]
Comment (cillascillas) [reply docs/introduction/getting-started.stories.mdx]: Discussed it with @icurdu and replaced it with an email address!

[the remark this answers is not in the document]
Comment (afontcu) [reply docs/introduction/getting-started.stories.mdx]: great!

File: docs/introduction/getting-started.stories.mdx (hunk 2 above)
Comment (cillascillas) [reply docs/introduction/getting-started.stories.mdx]: Same as in the comment below!

File: docs/introduction/welcome.stories.mdx (hunk 1 above)
Comment (icurdu) [reply docs/introduction/welcome.stories.mdx]: You are totally right! I owe you a beer to fix this mistake xD 
Can we add Lleida and Montgat?

File: docs/introduction/getting-started.stories.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (icurdu) [reply docs/introduction/getting-started.stories.mdx]: We can change the slack channels for productdesign@holaluz.com 😛

**Pull request merged**